x-pack/plugins/entity_manager/public/lib/entity_client.ts

```diff
@@ -87,6 +88,27 @@ export class EntityClient {
     }
   }
 
+  async getEntityDefinition(
+    id: string
+  ): Promise<{ definitions: EntityDefinition[] | EntityDefinitionWithState[] }> {
+    try {
+      return await this.repositoryClient(
+        'GET /internal/entities/definition/{id}' as unknown as keyof EntityManagerRouteRepository,
```

Comment:
@elastic/obs-entities If I use `GET /internal/entities/definition/{id?}` as in the type definition prefilling the id from the `params` is not working and as the type is generated we can't have both  `GET /internal/entities/definition/{id}` and  `GET /internal/entities/definition/{id?}` and in this case I want the id to be mandatory - that's why I use the typecasting (which I am not proud of, of course, but helps in this case) 

If we don't have the optional use case we can change it to `GET /internal/entities/definition/{id}` and have a different endpoint without the id in case we want to have it (`GET /internal/entities/definition` for example to handle the case and to make the usage easier, wdyt?

Reply:
1. Totally would prefer to have two endpoints (get all definitions vs get definition by ID)
2. I don't understand why you cannot use `GET /internal/entities/definition/{id?}`? The ID is mandatory in the client function `getEntityDefinition` even if the API itself says it's optional?

Reply:
I added a separate issue (#201789) for splitting the endpoint into 2.

Reply:
If we want to fix this in a follow up PR, can we then try to remove the casting in your PR and instead use a `@ts-expect-error` tag? So that we'll get a type failure once we change the APis?

Or, as I mentioned on Slack, we can fix the optional param handling in the [route repository package](https://github.com/elastic/kibana/blob/main/packages/kbn-server-route-repository-utils/src/format_request.ts#L17).

Reply:
I added a fix in [formatRequest helper](https://github.com/elastic/kibana/pull/201433/commits/784e9dc0b12a498e8487161b3697d36663a1f92f)

Reply:
I opened a follow-up issue ([#202151](https://github.com/elastic/kibana/issues/202151)) to improve the logic to support multiple params combinations as discussed offline and I will refine it, thanks for approving, I am resolving this one and I will merge